### PR TITLE
Move the metrics definition to pdata. Remove internal/data definition

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates #841

80% of the changes are deleting old data/metrics and copying the code to pdata/metrics. Unfortunately github does not see this as a code move.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Need to split this PR, will do tomorrow

File: consumer/pdata/metric.go [hunk 1]
```diff
@@ -35,10 +37,114 @@ func (at AggregationTemporality) String() string {
...
+
+// MetricDataFromOtlp creates the internal MetricData representation from the OTLP.
+func MetricsFromOtlp(orig []*otlpmetrics.ResourceMetrics) Metrics {
+	return Metrics{&orig}
+}
+
+// MetricDataToOtlp converts the internal MetricData to the OTLP.
+func MetricsToOtlp(md Metrics) []*otlpmetrics.ResourceMetrics {
+	return *md.orig
+}
```
Comment (tigrannajaryan) [inline consumer/pdata/metric.go:52]: Can we use the same approach as this https://github.com/open-telemetry/opentelemetry-collector/pull/1703 to make this functions limited for use in this repo only?

File: consumer/pdatautil/pdatautil.go [hunk 2]
```diff
@@ -19,69 +19,39 @@ package pdatautil
...
-	panic("Unsupported metrics type.")
+	return md.MetricCount()
 }
 
 func MetricAndDataPointCount(md pdata.Metrics) (int, int) {
-	if ims, ok := md.InternalOpaque.(data.MetricData); ok {
-		return ims.MetricAndDataPointCount()
-	}
-	panic("Unsupported metrics type.")
+	return md.MetricAndDataPointCount()
```
Comment (tigrannajaryan) [inline consumer/pdatautil/pdatautil.go:51]: Are these functions still needed?

File: consumer/pdata/metric.go (hunk 1 above)
Comment (bogdandrutu) [reply consumer/pdata/metric.go:52]: Will do in a separate PR.

File: consumer/pdatautil/pdatautil.go (hunk 2 above)
Comment (bogdandrutu) [reply consumer/pdatautil/pdatautil.go:51]: Yes because trying to make updating contrib easier, and limit the size of this PR. Will remove soon.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**